monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -1916,6 +1917,34 @@ sealed abstract class Task[+A] extends TaskBinCompat[A] with Serializable {
     */
   final def uncancelable: Task[A] =
     TaskCancellation.uncancelable(this)
+
+  /**
+    * TODO: doc
+    *
+    * @return
+    */
+  final def timedAttempt: Task[(Duration, Either[Throwable, A])] =
+    Task.deferAction { scheduler =>
+      val now: Task[Long] = Task.eval(scheduler.clockMonotonic(TimeUnit.NANOSECONDS))
```

Comment:
There should be a globally accessible instance of `Timer[Task]` or `Clock[Task]` (I'm not sure if we've caught up with changes to cats-effect, so there might not be, but it was the case previously), which should remove the need to wrap the thing into eval and maybe even use `deferAction` at all.

Reply:
It's not what @Avasil says here: https://github.com/monix/monix/issues/714#issue-358293580 or on Twitter.

The `Timer` thing was implemented in Cats-Effect after the Monix 3 RC1, I think. So, I'm not sure that this concept is implemented here, for now at least but I could be wrong. I don't know all the Monix code.

Reply:
Yeah, there's actually `Task.clock` method on a companion as a shortcut.

Here's a [test suite](https://github.com/monix/monix/blob/master/monix-eval/shared/src/test/scala/monix/eval/TaskClockTimerAndContextShiftSuite.scala#L30), you might get some hints from that.

Also, I think `clockRealTime` is more intended for measurements than `clockMonotonic`.

Reply:
I'll take a look ! Thanks for the help Oleg ! :)

Reply:
Actually, disregard the latest bit. I always mix the two up, sorry!

Reply:
> Actually, disregard the latest bit. I always mix the two up, sorry!

Like everyone I think :)